lib/call-cabal-project-to-nix.nix

```diff
@@ -1,5 +1,5 @@
 { dotCabal, pkgs, runCommand, nix-tools, cabal-install, ghc, hpack, symlinkJoin, cacert, index-state-hashes, haskellLib, materialize }@defaults:
-{ name          ? src.name or (baseNameOf src) # optional name for better error messages
+{ name          ? src.name or null # optional name for better error messages
 , src
```

Comment:
Using `baseNameOf` here has the same caching issues it did with `cleanSourceWith` and `cleanGit`